service/service.go

```diff
@@ -71,38 +71,17 @@ func newService(set *settings) (*service, error) {
 	if err = srv.telemetryInitializer.init(set.BuildInfo, srv.telemetrySettings.Logger, set.Config.Service.Telemetry, set.AsyncErrorChannel); err != nil {
 		return nil, fmt.Errorf("failed to initialize telemetry: %w", err)
 	}
-	srv.telemetrySettings.MeterProvider = srv.telemetryInitializer.mp
-
-	extensionsSettings := extensions.Settings{
-		Telemetry: srv.telemetrySettings,
-		BuildInfo: srv.buildInfo,
-		Configs:   srv.config.Extensions,
-		Factories: srv.host.factories.Extensions,
-	}
-	if srv.host.extensions, err = extensions.New(context.Background(), extensionsSettings, srv.config.Service.Extensions); err != nil {
-		return nil, fmt.Errorf("failed build extensions: %w", err)
-	}
 
```

Comment:
nit: The diff shows the `srv.telemetrySettings.MeterProvider = srv.telemetryInitializer.mp` line as changed, but really there is just an extra line here. Removing it should simplify the diff.